cmd/bucket-notification-handlers.go

```diff
@@ -146,10 +146,12 @@ func (api objectAPIHandlers) PutBucketNotificationHandler(w http.ResponseWriter,
 
 	rulesMap := config.ToRulesMap()
 	globalNotificationSys.AddRulesMap(bucketName, rulesMap)
-	for nerr := range globalNotificationSys.PutBucketNotification(bucketName, rulesMap) {
-		logger.GetReqInfo(ctx).AppendTags("remotePeer", nerr.Host.Name)
-		logger.LogIf(ctx, nerr.Err)
-	}
+	go func() {
```

Comment:
You would need to handle global signals properly.  IMO goroutine and signal handling needs to be inside `NotificationSys.PutBucketNotification()` itself than spelling the logic here.

Reply:
I have resent the PR, note that I have just moved logging to inside `globalNotificationSys.PutBucketNotification` from outside.